mlir/lib/Quantum/Transforms/MergeRotationsPatterns.cpp

```diff
@@ -322,6 +322,24 @@ struct MergeRotationsRewritePattern : public OpRewritePattern<OpType> {
     }
 };
 
+bool matchingQubitsAndPaulis(PPRotationArbitraryOp op, PPRotationArbitraryOp parentOp)
+{
+    // construct map
+    std::unordered_map<llvm::hash_code, Attribute> qubit_to_pauli;
+    for (auto [qubit, pauli] : llvm::zip(parentOp.getOutQubits(), parentOp.getPauliProduct())) {
+        qubit_to_pauli[hash_value(qubit)] = pauli;
+    }
+
+    // check pairings
+    for (auto [qubit, pauli] : llvm::zip(op.getInQubits(), op.getPauliProduct())) {
+        if (qubit_to_pauli[hash_value(qubit)] != pauli) {
+            return false;
+        }
+    }
```

Comment:
I think you can use `Value` directly

```suggestion
    std::unordered_map<Value, Attribute> qubit_to_pauli;
    for (auto [qubit, pauli] : llvm::zip(parentOp.getOutQubits(), parentOp.getPauliProduct())) {
        qubit_to_pauli[qubit] = pauli;
    }

    // check pairings
    for (auto [qubit, pauli] : llvm::zip(op.getInQubits(), op.getPauliProduct())) {
        if (qubit_to_pauli[qubit] != pauli) {
            return false;
        }
    }
```